src/client/Client.cc

```diff
@@ -6347,6 +6367,10 @@ int Client::mknod(const char *relpath, mode_t mode, const UserPerm& perms, dev_t
   tout(cct) << relpath << std::endl;
   tout(cct) << mode << std::endl;
   tout(cct) << rdev << std::endl;
+
+  if(std::string(relpath) == "/")
```

Comment:
Minor style point: please could you make all these if statements have a space before the opening ( like this "if (std::string..."